arangod/Cluster/ClusterComm.cpp

```diff
@@ -383,8 +386,22 @@ void ClusterComm::stopBackgroundThreads() {
   }  // for
 
   // pass 2:  verify each thread is stopped, wait if necessary
-  //          (happens in destructor)
+  //          No communication after this.
+  for (ClusterCommThread* thread : _backgroundThreads) {
+    if (!thread->runningInThisThread()) {
+      thread->halt();
```

Comment:
This looks really strange to me! IMHO it should never be possible, that this code is run by one of the background threads itself. Do you disagree?

Reply:
IMHO the only caller of `stopBackgroundThreads` should actually be the main thread